Components/TransactionReport/TransactionReport.php

```diff
@@ -0,0 +1,136 @@
+     *
+     * @return void
+     */
+    public function reportOrder($orderId)
+    {
+        $this->connection->createQueryBuilder()
+            ->insert(self::TRANSACTION_REPORT_TABLE)
+            ->setValue('order_id', ':orderId')
+            ->setParameter('orderId', $orderId)
+            ->execute();
+    }
+
+    /**
+     * @param string $shopwareVersion
+     *
+     * @return void
+     */
+    public function report($shopwareVersion, Client $client)
+    {
+        $reportResult = $this->getReportResult($this->getReportedOrderIds());
+        $currencies = $reportResult->getCurrencies();
+
+        foreach ($currencies as $currency) {
+            $requestBody = [
+                'identifier' => self::API_IDENTIFIER,
+                'reportDate' => (new DateTime())->format('Y-m-d\\TH:i:sP'),
+                'shopwareVersion' => $shopwareVersion,
+                'reportDataKeys' => ['turnover' => $reportResult->getTurnover($currency)],
+                'currency' => $currency,
+            ];
+
+            $client->post(self::POST_URL_ENDPOINT, [RequestOptions::JSON => $requestBody]);
+
+            $this->deleteReportedOrders($reportResult->getOrderIds($currency));
+        }
+    }
+
+    /**
+     * @return array<int, int>
+     */
+    private function getReportedOrderIds()
+    {
+        return $this->connection->createQueryBuilder()
+            ->select(['order_id'])
+            ->from(self::TRANSACTION_REPORT_TABLE)
+            ->execute()
+            ->fetchAll(PDO::FETCH_COLUMN);
+    }
+
+    /**
+     * @param array<int, int> $reportedOrderIds
+     *
+     * @return void
+     */
+    private function deleteReportedOrders(array $reportedOrderIds)
+    {
+        $this->connection->createQueryBuilder()
+            ->delete(self::TRANSACTION_REPORT_TABLE)
+            ->where('order_id IN (:reportedOrderIds)')
+            ->setParameter('reportedOrderIds', $reportedOrderIds, Connection::PARAM_INT_ARRAY)
+            ->execute();
+    }
+
+    /**
+     * @param array<int, int> $reportedIds
+     *
+     * @return ReportResult
+     */
+    private function getReportResult(array $reportedIds)
+    {
+        $queryBuilder = $this->connection->createQueryBuilder();
+
+        $result = $queryBuilder->select(['currency', 'id', 'invoice_amount'])
+            ->from('s_order')
+            ->where('id IN (:ordersToReport)')
+            ->andWhere('cleared = :paymentStatus')
+            ->andWhere('paymentID IN (:paymentIds)')
+            ->setParameter('paymentStatus', Status::PAYMENT_STATE_COMPLETELY_PAID)
+            ->setParameter('ordersToReport', $reportedIds, Connection::PARAM_INT_ARRAY)
+            ->setParameter('paymentIds', $this->getPaymentIds(), Connection::PARAM_INT_ARRAY)
+            ->execute()
+            ->fetchAll(PDO::FETCH_GROUP);
+
+        return new ReportResult($result);
+    }
+
+    /**
+     * @return array<int, int>
+     */
+    private function getPaymentIds()
```

Comment:
why not using the `\SwagPaymentPayPalUnified\Components\PaymentMethodProvider`?

Reply:
Unnecessary dependency